cpp/negentropy_wrapper.c

```diff
@@ -298,3 +282,153 @@ void free_result(result* r){
         free((void *)r->error);
     }
 }
+
+/*SubRange specific functions
+TODO: These and above methods need to be optimized to reduce code duplication*/
+
+void* subrange_new(void* storage, uint64_t startTimeStamp, uint64_t endTimeStamp){
+    negentropy::storage::BTreeMem* st = reinterpret_cast<negentropy::storage::BTreeMem*>(storage);
+    negentropy::storage::SubRange* subRange = NULL;
+    try {
+    subRange = new negentropy::storage::SubRange(*st, negentropy::Bound(startTimeStamp), negentropy::Bound(endTimeStamp));
+    } catch (negentropy::err e){
+        return NULL;
```

Comment:
We need to enhance the error detail. We can't just catch and forget, can we :P ?

Reply:
Makes sense to return error back..will make this change.